backend/app/Repositories/EloquentSessionRepository.php

```diff
@@ -38,4 +36,10 @@ public function getAvgSession(AverageSessionFilter $filter): Collection
                 ->where('start_session', '<=', $filter->getEndDate())
                 ->get();
     }
+
+    public function lastActiveByVisitorId(int $visitorId): ?Session
+    {
+        return Session::whereVisitorId($visitorId)
+            ->whereTime('updated_at', '>', (Carbon::now())->subMinutes(30)->toDateTimeString())->first();
```

Comment:
if I am not mistaken, visitor has particular property instead of updated_at: last_activity

Reply:
and still, visitor has particular property last_activity

Reply:
Okay, I see, it should be end_session so, as it the same as last_activity of session. Or date of last session visit